### ISSUE TITLE
Repeated values using TwitterPager with Counts Endpoint

### ISSUE THREAD

**Issue by rsbensaid** [geduldig/TwitterAPI#116]
Hi there - I'm running into an issue using TwitterPager to access the fullarchive counts endpoint. As stated in the title, the issue is that I see the same values coming back with each page.

I am not seeing any issue when I use TwitterPager to access the search endpoint, only seeing this issue with counts.

Here is my code: 
`r = TwitterPager(api, req_uri, {'query': query_string
		, 'fromDate':'201806010000'
		, 'toDate':'201812010000'
		, 'bucket': 'day'
	})`

`	tweet_count = 0`
`for item in r.get_iterator(new_tweets = False):`
`		tweet_count += item['count']`
`
		print item['timePeriod'], item['count']
`

and here is the repeated output I'm seeing: 

"""
201810310000 14155
201811010000 6439
201811020000 4929
201811030000 5225
201811040000 2521
201811050000 5065
201811060000 15878
201811070000 47956
201811080000 10732
201811090000 9390
201811100000 9950
201811110000 12128
201811120000 6284
201811130000 8060
201811140000 6480
201811150000 13646
201811160000 6863
201811170000 16886
201811180000 4656
201811190000 8002
201811200000 14489
201811210000 14061
201811220000 10976
201811230000 10600
201811240000 6619
201811250000 2362
201811260000 3297
201811270000 6453
201811280000 17963
201811290000 9314
201811300000 2288
201810310000 14155
201811010000 6439
201811020000 4930
201811030000 5225
201811040000 2521
201811050000 5065
201811060000 15878
201811070000 47956
201811080000 10732
201811090000 9390
201811100000 9949
201811110000 12128
201811120000 6284
201811130000 8060
201811140000 6480
201811150000 13647
201811160000 6863
201811170000 16886
201811180000 4656
201811190000 8002
201811200000 14489
201811210000 14060
201811220000 10976
201811230000 10600
201811240000 6619
201811250000 2362
201811260000 3297
201811270000 6453
201811280000 17963
201811290000 9315
201811300000 2288
"""

I only added the 'new_tweets' argument in get_iterator because I thought flipping that to True might somehow affect pagination - I got repeated results again, just in the opposite order :) 

Thanks in advance for your help and sorry if I've missed something here and the issue is on my end!

**Comment by geduldig** [comment]
I'll take a look. Thanks!

**Comment by geduldig** [comment]
I'm testing in a sandbox environment. I don't have access to the count feature, so I can't repeat exactly what you are doing. For me the `bucket` parameter is not supported.

Can you try removing the `bucket` parameter from your query and printing `item['created_at']`?
When I do this the output seems correct in that the dates do not repeat. If this works for you then we know it's something do to with the output results in a non-sandbox environment. Then maybe I can get access to test.

**Comment by rsbensaid** [comment]
Awesome! I'd be happy to test when ready 👍

**Comment by geduldig** [comment]
It looks like the request you are making does not return pages of results. I think it returns all the results in one long array. If that is the case, you wouldn't use the `TwitterPager` helper class. Simply make the request like this:

```python
r = api.request(
    req_uri,
    {'query': query_string , 'fromDate':'201806010000' , 'toDate':'201812010000' , 'bucket': 'day' })
```

**Comment by rsbensaid** [comment]
Hmm, are you sure this type of request won't return pages? I'd assumed it would as one request to the counts API will return values for a max of 31 days and the docs say getting more requires a 'next' token just like the 'data' endpoint. 

From twitter's docs: "
**Counts pagination**
The 'counts' endpoint provides Tweet volumes associated with a query on either a daily, hourly, or per-minute basis. The 'counts' API endpoint will return a timestamped array of counts for a maximum of a 31-day payload of counts.

For higher volume queries, there is the potential that the generation of counts will take long enough to potentially trigger a response timeout. When this occurs you will receive less than 31 days of counts but will be provided a 'next' token in order to continue making requests for the entire payload of counts.

As with the data 'next' tokens, you must make the exact same query as the original and also include a 'next' request parameter set to the value from the previous response.


...continued...

Using only the fromDate with no toDate parameter, the API will deliver counts (data volumes) data for the query going back in time from now until the fromDate. If the fromDate is older than 31 days from now( ), you will receive a 'next' token to page through your request.
"

 For example, the request you detailed above returns, but it only has values for 10/31/2018 - 11/30/2018.

Either way - thanks for your replies here, I've just gone ahead and put in a workaround changing the fromDate and toDate values in increments of 31 days with each request to achieve the desired result.

**Comment by geduldig** [comment]
Ahh, yes, you are correct. I missed that...

**Comment by geduldig** [comment]
Okay, should be fixed now in version 2.5.8. Let me know if you still have issues. Thanks!
